File: php/features.php
<?php
declare(strict_types=1);

// BluefinDecryptx SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class BluefinDecryptxFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new BluefinDecryptxBaseFeature();
            case "test":
                return new BluefinDecryptxTestFeature();
            default:
                return new BluefinDecryptxBaseFeature();
        }
    }
}

File: py/features.py
# BluefinDecryptx SDK feature factory

from feature.base_feature import BluefinDecryptxBaseFeature
from feature.test_feature import BluefinDecryptxTestFeature


def _make_feature(name):
    features = {
        "base": lambda: BluefinDecryptxBaseFeature(),
        "test": lambda: BluefinDecryptxTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
